lib/mail/fields/common_date_field.rb

```diff
@@ -4,26 +4,32 @@
 
 module Mail
   class CommonDateField < NamedStructuredField #:nodoc:
-    def self.singular?
-      true
-    end
+    class << self
+      def singular?
+        true
+      end
 
-    def self.normalize_datetime(string)
-      if Utilities.blank?(string)
-        datetime = ::DateTime.now
-      else
-        stripped = string.to_s.gsub(/\(.*?\)/, '').squeeze(' ')
-        begin
-          datetime = ::DateTime.parse(stripped)
-        rescue ArgumentError => e
-          raise unless 'invalid date' == e.message
+      def normalize_datetime(string)
+        if Utilities.blank?(string)
+          datetime = ::DateTime.now
+        else
+          stripped = string.to_s.gsub(/\(.*?\)/, '').squeeze(' ')
+          with_invalid_date_time_error_handling do
+            datetime = ::DateTime.parse(stripped)
+          end
+        end
+
+        if datetime
+          datetime.strftime('%a, %d %b %Y %H:%M:%S %z')
+        else
+          string
         end
       end
 
-      if datetime
-        datetime.strftime('%a, %d %b %Y %H:%M:%S %z')
-      else
-        string
+      def with_invalid_date_time_error_handling(&block)
+        yield
+      rescue ArgumentError => e
+        raise unless e.message =~ /\A(invalid date|string length \(\d+\) exceeds the limit \d+)\z/
```

Comment:
Perhaps this should live in its own method (a method to parse dates that return `nil` when invalid or too long)  in `Mail::Utilities` so it's used from here and from `ReceivedElement`.

Reply:
Yes, please move it into utilities.